src/big/json.cr

```diff
@@ -1,44 +1,81 @@
 require "json"
 require "big"
 
-def BigInt.new(pull : JSON::PullParser)
-  pull.read_int
-  BigInt.new(pull.raw_value)
-end
+struct BigInt
+  def self.new(pull : JSON::PullParser)
+    case pull.kind
+    when .int?
+      pull.read_int
+      value = pull.raw_value
+    else
+      value = pull.read_string
```

Comment:
I don't think a string should be implicitly converted to a big int... we don't do that for regular ints, why should we do it for BigInt?